mypy/checkexpr.py

```diff
@@ -1061,15 +1061,51 @@ def check_op(self, method: str, base_type: Type, arg: Node,
             # method of the other operand instead.
             rmethod = self.get_reverse_op_method(method)
             arg_type = self.accept(arg)
+            base_arg_node = TempNode(base_type)
+            user_error = None  # type: Tuple[Tuple[Type, Type], MessageBuilder]
             if self.has_member(arg_type, rmethod):
-                method_type = self.analyze_external_member_access(
-                    rmethod, arg_type, context)
-                temp = TempNode(base_type)
-                return self.check_call(method_type, [temp], [nodes.ARG_POS],
-                                       context)
+                local_errors = self.msg.copy()
```

Comment:
This code gets repeated almost identically three times. Maybe define a nested function that avoids the repetition? It would probably also make the function shorter -- it's getting pretty long and somewhat hard to follow.